mapper-processor/src/main/java/com/datastax/oss/driver/internal/mapper/processor/SaveQueryGenerator.java

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright DataStax, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.datastax.oss.driver.internal.mapper.processor;
+
+import com.datastax.oss.driver.api.core.cql.BoundStatement;
+import com.datastax.oss.driver.api.core.cql.PrepareRequest;
+import com.datastax.oss.driver.api.core.cql.PreparedStatement;
+import com.datastax.oss.driver.api.core.cql.Statement;
+import com.datastax.oss.driver.api.mapper.annotations.SaveQuery;
+import com.datastax.oss.driver.api.mapper.annotations.Table;
+import com.datastax.oss.driver.api.querybuilder.QueryBuilderDsl;
+import com.datastax.oss.driver.internal.core.cql.DefaultPrepareRequest;
+import com.squareup.javapoet.ClassName;
+import com.squareup.javapoet.FieldSpec;
+import com.squareup.javapoet.MethodSpec;
+import com.squareup.javapoet.TypeSpec;
+import java.util.List;
+import javax.lang.model.element.Element;
+import javax.lang.model.element.ElementKind;
+import javax.lang.model.element.ExecutableElement;
+import javax.lang.model.element.Modifier;
+import javax.lang.model.element.TypeElement;
+import javax.lang.model.element.VariableElement;
+import javax.lang.model.type.TypeMirror;
+
+public class SaveQueryGenerator implements PartialClassGenerator {
+
+  private final String methodName;
+  private final String parameterName;
+  private final TypeMirror parameterType;
+  private final String preparedStatementFieldName;
+  private final String tableName;
+  private final EntityDefinition entityDefinition;
+
+  public SaveQueryGenerator(
+      ExecutableElement methodElement, SaveQuery annotation, GenerationContext context) {
+    this.methodName = methodElement.getSimpleName().toString();
+    preparedStatementFieldName = methodName + "_preparedStatement";
+    tableName = annotation.table();
+    List<? extends VariableElement> parameters = methodElement.getParameters();
+    if (parameters.size() != 1) {
+      context.getMessager().error(methodElement, "Expected a single argument");
+      throw new SkipGenerationException();
+    }
+    VariableElement parameter = parameters.get(0);
+    parameterName = parameter.getSimpleName().toString();
+    parameterType = parameter.asType();
+    entityDefinition = context.getEntityDefinitions().get(parameterType);
+    if (entityDefinition == null) {
+      context.getMessager().error(methodElement, "Missing definition of entity %s", parameterType);
+      throw new SkipGenerationException();
+    }
+  }
```

Comment:
in generated `save` we are not binding and extracting parameters from Product

Reply:
I'm working on some changes so I don't have access to the generated code right now, but that should get handled by the loop at line 113.